model/cloudfoundry/registry.yaml

```diff
@@ -0,0 +1,157 @@
+groups:
+  - id: registry.cloudfoundry
+    type: attribute_group
+    brief: >
+      CloudFoundry resource attributes.
+    attributes:
+      - id: cloudfoundry.system.id
+        type: string
+        stability: experimental
+        brief: >
+          A guid or another name describing the event source.
+        note: |
+          CloudFoundry defines the `source_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the component name, e.g. "gorouter", for
+          CloudFoundry components.
+
+          When system components are instrumented, values from the
+          [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `system.id` should be set to
+          `spec.deployment/spec.name`.
+        examples:
+          - 'cf/gorouter'
+      - id: cloudfoundry.system.instance.id
+        type: string
+        stability: experimental
+        brief: >
+          A guid describing the concrete instance of the event source.
+        note: |
+          CloudFoundry defines the `instance_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the vm id for CloudFoundry components.
+
+          When system components are instrumented, values from the
+          [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `system.instance.id` should be set to `spec.id`.
+        examples:
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+      - id: cloudfoundry.app.name
+        type: string
+        stability: experimental
+        brief: >
+          The name of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_name`. This is the same value
+          as reported by `cf apps`.
+        examples: ['my-app-name']
+      - id: cloudfoundry.app.id
+        type: string
+        stability: experimental
+        brief: >
+          The guid of the application.
+        note: |
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_id`. This is the same value as
+          reported by `cf app <app-name> --guid`.
+        examples: ['218fc5a9-a5f1-4b54-aa05-46717d0ab26d']
+      - id: cloudfoundry.app.instance.id
+        type: string
```

Comment:
is it intentionally a `string`? Examples and description suggest it could be an `int`. If you want to have string for extensibility/edge cases, it's totally fine too.

Reply:
Thanks for that finding. Since we separated the `application.instance.id` from the `system.instance.id`, this could indeed be an `int`. Still I would prefer a `string` exactly for extensibility and edge cases.